### DISCUSSION TITLE
Saleor Tutorial for filter: {search: "Blue"} does not output the result to the page

### DISCUSSION THREAD

**Discussion by amityweb** [saleor/saleor#9732] [General]
I am following the Saleor tutorial. Does anyone know why the first query works, but the second one with filter search shows nothing? If I enter this query into the Saleor Core Query playground it works fine because there is a product that matches "Blue". I expect to see 1 result with the word blue in it, just like in the query playground, but nothing is showing, its an empty < ul > block. It happens using any keyword in any of the Saleor example products. Query playground shows a product returned. 

**Works**
```
query FetchTwelveProducts {
	products(first: 12, channel: "default-channel") {
		edges {
			node {
				id 
				name
				thumbnail {
					url
				}
				category {
					name
				}
			}
		}
	}
}
```

**The following does not work**, it does not output the results even though the query in playground works fine and shows the exact same product structure compared to if I fetch just 1 product using the first query above:

```
query FetchTwelveProducts {
	products(first: 12, channel: "default-channel", filter: { search: "Blue" }) {
		edges {
			node {
				id 
				name
				thumbnail {
					url
				}
				category {
					name
				}
			}
		}
	}
}
```

Here is the ProductCollection.tsx code as per Saleor tutorial:
```
export const ProductCollection = () => {
  const { loading, error, data } = useFetchTwelveProductsQuery();

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error</p>;

  if (data) {
    const products = data.products?.edges || [];

    return (
      <ul role="list" className={styles.grid}>
        {products?.length > 0 &&
          products.map(
            ({ node }) => <ProductElement key={node.id} {...node as Product} />,
          )}
      </ul>
    );
  }

  return null;
```

**Comment by zaiste** [comment]
@amityweb could you share you repository?

**Reply by amityweb** [reply]
Thank you, I have created a temporary copy of the repo to add you and sent an invite as its private
